calculation/spatial_hazard/spatial_hazard/correlate_ims.py

```diff
@@ -0,0 +1,227 @@
+    result_dict = {station: [] for station in stations}
+    for imdb_ffp in imdb_ffps:
+        print("Getting {}".format(imdb_ffp))
+        with gc.dbs.IMDB.get_imdb(imdb_ffp, writeable=False) as imdb:
+            imdb_stations = imdb.sites().index
+            for station in stations:
+                if station not in imdb_stations:
+                    print("Station {} has no entries here".format(station))
+                cur_data = imdb.im_data(station, im)
+                if cur_data is not None:
+                    result_dict[station].append(cur_data)
+
+    result_dict = {key: pd.concat(value) for key, value in result_dict.items()}
+    return result_dict
+
+
+def load_stations_fault_data(
+    imdb_ffps: Sequence[str], stations: Sequence[str], im: gc.im.IM, fault: str
+):
+    """Loads the IM data for the specified stations and fault from the IMDBs"""
+    stations_data = load_stations_data(imdb_ffps, stations, im)
+
+    return pd.DataFrame(
+        {
+            cur_station: cur_df.loc[fault].to_dict()
+            for cur_station, cur_df in stations_data.items()
+        }
+    ).T
+
+
+def calculate_distance_matrix(
+    stations: Sequence[str], locations_df: pd.DataFrame
+) -> pd.DataFrame:
+    """
+    Given a set of stations and their locations (in lat, lon format), calculate the matrix containing
+    the pairwise distance between each one
+
+    Parameters
+    ----------
+    stations: List[str]
+        List of the stations
+    locations_df: pd.DataFrame
+        Locations of each of the stations (in lat, lon)
+
+    Returns
+    -------
+
+    """
+    distance_matrix = -np.ones((len(stations), len(stations)))
+    for i, station_1 in enumerate(stations):
+        cur_dist, _ = calc_rrup_rjb(
+            np.asarray(
+                [[locations_df.loc[station_1].lon, locations_df.loc[station_1].lat, 0]]
+            ),
+            np.stack(
+                (
+                    locations_df.loc[stations].lon,
+                    locations_df.loc[stations].lat,
+                    np.zeros(len(stations)),
+                ),
+                axis=1,
+            ),
+        )
+        distance_matrix[i, :] = cur_dist
+    return pd.DataFrame(index=stations, data=distance_matrix, columns=stations)
+
+
+def get_corr_matrix(stations, distance_matrix, selected_im, emp_df):
+    """
+    Computes the correlation matrix for the specified stations
+    using the Loth & Baker (2013) model.
+
+    Parameters
+    ----------
+    stations: sequence of strings
+    distance_matrix: dataframe
+    selected_im: IM
+    emp_df: dataframe
+
+    Returns
+    -------
+    R: dataframe
+        The correlation matrix of the given station
+    """
+    assert np.all(
+        distance_matrix.index.values == emp_df.index.values
+    ), "Order of the stations has to be the same"
+
+    # Compute correlation matrix
+    R = np.eye(len(stations))
+    for i, station_i in enumerate(stations):
```

Comment:
same here, unlike above, this is now `station_i` but can this also be a `station`?